### PR TITLE
Fix tensor valued GPyTorch constraint processing

### PR DESCRIPTION
**Pull request by saitcakmak**
Summary: Fixes the issue reported in https://github.com/pytorch/botorch/discussions/1813#discussioncomment-5977542

Differential Revision: D46115436

### CODE REVIEW

File: test/optim/utils/test_numpy_utils.py [hunk 1]
```diff
@@ -106,7 +106,7 @@ def test_set_tensors_from_ndarray_1d(self):
             )
             n += k
 
-    def test_get_bounds_as_ndarray(self):
+    def ftest_get_bounds_as_ndarray(self):
```
Comment (saitcakmak) [inline test/optim/utils/test_numpy_utils.py:109]:
```suggestion
    def test_get_bounds_as_ndarray(self):
```

File: botorch/optim/utils/numpy_utils.py [hunk 2]
```diff
@@ -151,18 +151,17 @@ def get_bounds_as_ndarray(
         An ndarray of bounds.
     """
     inf = float("inf")
-    out = None
+    full_size = sum(prod(param.shape) for param in parameters.values())
```
Comment (Balandat) [inline botorch/optim/utils/numpy_utils.py:154]:
```suggestion
    full_size = sum(param.numel() for param in parameters.values())
```

File: botorch/optim/utils/numpy_utils.py [hunk 3]
```diff
@@ -151,18 +151,17 @@ def get_bounds_as_ndarray(
...
+    out = np.full((full_size, 2), (-inf, inf))
     index = 0
     for name, param in parameters.items():
         size = prod(param.shape)
```
Comment (Balandat) [inline botorch/optim/utils/numpy_utils.py:158]:
```suggestion
        size = param.numel()
```

**Review by Balandat — APPROVED** [review]
thanks for the quick fix!

**Pull request closed without merging**